contract/contracts/TokenFactory.sol

```diff
@@ -45,15 +47,16 @@ contract TokenFactory is TokenFactoryInterface {
         bool isTotalSupplyFixed,
         uint8 lockupPeriod, // years
         bool enableStakeToToken
-    ) public override returns (address) {
-        // TODO Validate ratio is between 0 and 100
+    ) external override returns (address) {
+        require(ratio <= 100, "Ratio must be a number between 0 and 100");
         // TODO register token to staking token list
         address tokenAddress;
-        { // To avoid stack too deep
+        {// To avoid stack too deep
             FanToken newToken = new FanToken(name, symbol, totalSupply, payable(address(this)), decimals, stakingPool);
             tokenAddress = address(newToken);
-            // TODO: Send token to vesting contract(for creator) and creator(for distribution to fans) depend on ratio
-            newToken.transfer(creator, totalSupply);
+            newToken.transfer(vesting, totalSupply.mul(ratio).div(100));
```

Comment:
Discussion:
If the totalSupply is less than 10, an error will occur.
Do you tolerate less than 10 totalSupply?

Reply:
Only 10 totalSupply token is unusual. So IMO we can exclude in such case.